spv/spv.go

```diff
@@ -1,23 +1,24 @@
-package paymail
+package spv
 
 import (
 	"context"
 	"errors"
-	"fmt"
 
 	"github.com/libsv/go-bt/v2"
-	"github.com/libsv/go-bt/v2/bscript/interpreter"
+
+	"github.com/bitcoin-sv/go-paymail"
 )
 
 type MerkleRootVerifier interface {
 	VerifyMerkleRoots(
 		ctx context.Context,
-		merkleRoots []MerkleRootConfirmationRequestItem,
+		merkleRoots []paymail.MerkleRootConfirmationRequestItem,
 	) error
 }
 
 // ExecuteSimplifiedPaymentVerification executes the SPV for decoded BEEF tx
-func ExecuteSimplifiedPaymentVerification(dBeef *DecodedBEEF, provider MerkleRootVerifier) error {
+func ExecuteSimplifiedPaymentVerification(dBeef *paymail.DecodedBEEF, provider MerkleRootVerifier) error {
```

Comment:
IMHO we should make SPV for each ancestor until the one which is mined, otherwise there is a breach that can be used to fool us.

Reply:
Looks you're right

Reply:
done